### DISCUSSION TITLE
`yarn link` a monorepo package which depends on other versioned monorepo packages

### DISCUSSION THREAD

**Discussion by alecmev** [yarnpkg/berry#5209] [Q&A] [answered]
I have a monorepo like this:

```
monorepo/
  packages/
    foo/
    bar/
```

Both packages are versioned and published. `foo` has `bar` listed in its dependencies, version `^1.0.0`.

Then I have a project outside of this monorepo, and it depends on `foo`. I want to do `yarn link ~/monorepo/packages/foo` for local development, but then Yarn pulls `bar` from our registry, rather than using the local version.

How do I instruct Yarn to resolve `bar` to `~/monorepo/packages/bar`? Write it explicitly in resolutions? Any way to have `yarn link` do that automatically? This matters because we link and unlink regularly.

Thanks!

**Comment by merceyz** [comment, marked as answer]
Use the `--all` flag to link the entire project https://yarnpkg.com/cli/link#options-A%2C-all.

**Reply by alecmev** [reply]
Thank you! Should've double-checked the docs first.
